library/src/main/java/com/dslplatform/json/JsonWriter.java

```diff
@@ -413,16 +413,7 @@ public final void writeAscii(final String value, final int len) {
 	 * @param buf byte buffer to copy
 	 */
 	public final void writeAscii(final byte[] buf) {
-		final int len = buf.length;
-		if (position + len >= buffer.length) {
-			enlargeOrFlush(position, len);
-		}
-		final int p = position;
-		final byte[] _result = buffer;
-		for (int i = 0; i < buf.length; i++) {
```

Comment:
The reason why this didn't redirect to the overloaded method is to simplify array bounds check elimination.
I guess Java could still do that when it inlines method in here, but it's non obvious to me that it will.

Reply:
They are small and I'd expect them to get inlined TBH.